### PR TITLE
add paligemma2 blog

### PR DESCRIPTION
**Pull request by merveenoyan**
@pcuenca @ariG23498

### CODE REVIEW

File: assets/paligemma/Paligemma2.png
Comment (pcuenca) [inline assets/paligemma/Paligemma2.png:1]: not loading for me

File: paligemma2.md [hunk 1]
```diff
@@ -0,0 +1,228 @@
+---
+title: "Welcome PaliGemma 2 – New vision language models by Google"
+thumbnail: /blog/assets/paligemma/paligemma2.png
```
Comment (pcuenca) [inline paligemma2.md:3]: this one is lowercase

File: paligemma2.md [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+- user: pcuenq
+- user: ariG23498
+---
+
+
+# Welcome PaliGemma 2 – New vision language models by Google
+
+We are excited to welcome Google's all-new vision language models, PaliGemma 2, a new iteration of PaliGemma. Like its predecessor, PaliGemma 2 uses the same powerful [SigLIP](https://huggingface.co/collections/google/siglip-659d5e62f0ae1a57ae0e83ba) for vision, but it upgrades to the latest Gemma 2 for the text decoder part.
+
+PaliGemma 2 comes with new pre-trained (pt) models in sizes 3B, 10B, and 28B, all of them supporting various input resolutions: 224x224, 448x448, and 896x896. These combinations provide a lot of flexibility for different use cases, so practitioners can choose the balance they need in the quality / efficiency space. In contrast, the previous PaliGemma was only available in the 3B variant.
```
Comment (pcuenca) [inline paligemma2.md:18]:
```suggestion
PaliGemma 2 comes with new pre-trained (pt) models, in sizes of `3B`, `10B`, and `28B` parameters. All of them support various input resolutions: `224x224`, `448x448`, and `896x896`. These combinations provide a lot of flexibility for different use cases, so practitioners can choose the balance they need in the quality / efficiency space. In contrast, the previous PaliGemma was only available in the 3B variant.
```

File: paligemma2.md [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+- [Fine-tuning Script](https://github.com/merveenoyan/smol-vision/blob/main/Fine_tune_PaliGemma.ipynb)
+  
+- [Demo for Fine-tuned Model](https://huggingface.co/spaces/merve/paligemma2-vqav2)
+  
+- [The technical report](https://huggingface.co/papers/2412.03555)
+
+
+## Table of Content
+
+* [Intro PaliGemma 2](#introducing-paligemma-2)
```
Comment (pcuenca) [inline paligemma2.md:39]:
```suggestion
* [Introducing PaliGemma 2](#introducing-paligemma-2)
```

File: paligemma2.md (hunk 1 above)
Comment (merveenoyan) [reply paligemma2.md:3]: https://github.com/huggingface/blog/pull/2515/files#diff-8fdc8d2fd997f719e0c90fdef4803e8ba776bc8831d75d559ce8e748f2d7f4e4 here it's capitalized

File: paligemma2.md [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+
+![PaliGemma2 Architecture](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/blog/paligemma/paligemma2_arch.png)
+
+The new models are based on the Gemma 2 2B, 9B, and 27B language models, resulting in the corresponding 3B, 10B, and 28B PaliGemma 2 variants, whose names take into account the additional parameters of the (compact) image encoder. As mentioned above, they support three different resolutions, providing great flexibility for fine-tuning to downstream tasks.
+
+PaliGemma 2 is distributed under the Gemma license, which allows for redistribution, commercial use, fine-tuning and creation of model derivatives.
+
+This release comes with the following checkpoints in `bfloat16` precision:
+
+- 9 pre-trained models: Pre-trained models in 3B, 10B, and 28B with resolutions of `224x224`, `448x448`, and `896x896`.
```
Comment (pcuenca) [inline paligemma2.md:63]:
```suggestion
- 9 pre-trained models: 3B, 10B, and 28B with resolutions of `224x224`, `448x448`, and `896x896`.
```

File: paligemma2.md [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+  
+## Model Capabilities
+
+As seen with the previous PaliGemma release, the pre-trained (pt) models work great for further fine-tuning on downstream tasks. 
+
+The pt models are pre-trained on the following data mixture. The diversity of the pre-training dataset allows fine-tuning on downstream tasks in similar domains to be carried out using comparatively fewer examples.
+
+- **WebLI**: A web-scale multilingual image-text dataset built from the public web. A wide range of WebLI splits is used to acquire versatile model capabilities, such as visual semantic understanding, object localization, visually-situated text understanding, and multilinguality.
+  
+- **CC3M-35L:** Curated English image-alt_text pairs from webpages [Sharma et al., 2018](https://aclanthology.org/P18-1238/). To label this dataset, the authors used [Google Cloud Translation API](https://cloud.google.com/translate) to translate into 34 additional languages.
```
Comment (pcuenca) [inline paligemma2.md:75]:
```suggestion
- **CC3M-35L:** Curated English image-alt_text pairs from webpages ([Sharma et al., 2018](https://aclanthology.org/P18-1238/)). To label this dataset, the authors used [Google Cloud Translation API](https://cloud.google.com/translate) to translate into 34 additional languages.
```

File: paligemma2.md [hunk 6]
```diff
@@ -0,0 +1,228 @@
...
+  
+- **Visual Question Generation with Question Answering Validation” (VQ2A):** An improved dataset for question answering. The dataset is translated into the same additional 34 languages, using the Google Cloud Translation API.
```
Comment (pcuenca) [inline paligemma2.md:77]:
```suggestion
- **Visual Question Generation with Question Answering Validation (VQ2A):** An improved dataset for question answering. The dataset is translated into the same additional 34 languages, using the Google Cloud Translation API.
```

File: paligemma2.md [hunk 7]
```diff
@@ -0,0 +1,228 @@
...
+If you have previously fine-tuned PaliGemma, the API to fine-tune PaliGemma2 is the same, you can use your code out of the box. We provide a fine-tuning script and a notebook for you to fine-tune the model, freeze parts of the model, or apply memory efficient fine-tuning techniques like LoRA or QLoRA.
+
+We have LoRA fine-tuned model on half of the VQAv2 validation split for demonstration purposes. This took half an hour on 3 A100s with 80GB VRAM. The model can be found [here](https://huggingface.co/merve/paligemma2-3b-vqav2) and the demo can be found [here](https://huggingface.co/spaces/merve/paligemma2-vqav2).
+
+
+## Conclusion
+
+The new PaliGemma 2 release is even more exciting than the previous one with various sizes fitting everyone’s needs. We are looking forward to see what the community is going to build off of the even more powerful PaliGemma 2 pre-trained models.
+
+We thank Google team for releasing this amazing model family. Big thanks to Pablo Montalvo for integrating the model to transformers.
```
Comment (ariG23498) [inline paligemma2.md:218]:
```suggestion
We thank Google team for releasing this amazing model family. Big thanks to [Pablo Montalvo](https://huggingface.co/Molbap) for integrating the model to transformers.
```

File: paligemma2.md [hunk 8]
```diff
@@ -0,0 +1,228 @@
...
+
+- `bfloat16`, no quantization: 60.04% accuracy.
+- `8-bit`: 59.78%.
+- `4-bit`, using the configuration from the snippet above: 58.72%.
+
+These are very encouraging figures! Of course, quantization is most interesting for the larger checkpoints, we recommend you always measure results on the domains and tasks you’ll be using.
+
+## Fine-tuning
+
+If you have previously fine-tuned PaliGemma, the API to fine-tune PaliGemma2 is the same, you can use your code out of the box. We provide a fine-tuning script and a notebook for you to fine-tune the model, freeze parts of the model, or apply memory efficient fine-tuning techniques like LoRA or QLoRA.
```
Comment (ariG23498) [inline paligemma2.md:209]:
```suggestion
If you have previously fine-tuned PaliGemma, the API to fine-tune PaliGemma 2 is the same, you can use your code out of the box. We provide a fine-tuning script and a notebook for you to fine-tune the model, freeze parts of the model, or apply memory efficient fine-tuning techniques like LoRA or QLoRA.
```

Also should we add notebook link embedded in this para?

File: paligemma2.md [hunk 9]
```diff
@@ -0,0 +1,228 @@
...
+  
+OpenImages: Detection and object-aware questions and answers (Piergiovanni et al. 2022) generated by handcrafted rules on the [OpenImages dataset](https://storage.googleapis.com/openimages/web/factsfigures_v7.html).
+- **WIT**: Images and texts collected from Wikipedia (Srinivasan et al., 2021). 
+
+The PaliGemma 2 team internally fine-tuned the PT models on a wide variety of visual-language understanding tasks, and they provide benchmarks of these fine-tuned models [in the model card](https://huggingface.co/google/paligemma2-28b-pt-896#paligemma-2-results-by-model-resolution-and-size) and [the technical report](https://huggingface.co/papers/2412.03555).
+
+PaliGemma 2 fine-tuned on the DOCCI dataset, can accomplish a wide range of captioning tasks, including text rendering, capturing spatial relations, and including world knowledge in captions.
+
+You can find the performance of the DOCCI fine-tuned PaliGemma 2 checkpoints, compared with other models (taken from Table 6 in [the technical report](https://huggingface.co/papers/2412.03555)).
```
Comment (pcuenca) [inline paligemma2.md:86]:
```suggestion
You can find below the performance of the DOCCI fine-tuned PaliGemma 2 checkpoints, compared with other models (taken from Table 6 in [the technical report](https://huggingface.co/papers/2412.03555)).
```

File: paligemma2.md [hunk 10]
```diff
@@ -0,0 +1,228 @@
...
+For demonstration purposes, we in the Hugging Face team fine-tuned [PaliGemma 2 3B with 448x448 resolution](https://huggingface.co/google/paligemma2-3b-pt-448) on a small portion of the [VQAv2 dataset](https://huggingface.co/datasets/merve/vqav2-small). We used LoRA fine-tuning and PEFT, as explained later in the fine-tuning section. The demo below showcases the final result. Feel free to examine the code in [the Space](https://huggingface.co/spaces/merve/paligemma2-vqav2) to see how it works, or clone it to adapt to your own fine-tunes.
+<figure class="image flex flex-col items-center text-center m-0 w-full">
+  <video alt="paligemma.mp4" autoplay loop autobuffer muted playsinline>
+    <source src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/blog/paligemma/paligemma2.mp4" type="video/mp4">
+  </video>
+  <figcaption></figcaption>
+</figure>
+
+## How to Use with Transformers
+You can run inference on the PaliGemma 2 models with 🤗 transformers, using the PaliGemmaForConditionalGeneration and AutoProcessor APIs. Until a PyPi version of transformers is released, you need to install it from the main branch as follows:
```
Comment (ariG23498) [inline paligemma2.md:155]:
```suggestion
## How to Use with Transformers

You can run inference on the PaliGemma 2 models with 🤗 transformers, using the PaliGemmaForConditionalGeneration and AutoProcessor APIs. Until a PyPi version of transformers is released, you need to install it from the main branch as follows:
```

File: paligemma2.md [hunk 11]
```diff
@@ -0,0 +1,228 @@
...
+raw_image = Image.open(requests.get(image_file, stream=True).raw)
+inputs = processor(prompt, raw_image, return_tensors="pt").to(“cuda”)
+output = model.generate(**inputs, max_new_tokens=20)
+
+print(processor.decode(output[0], skip_special_tokens=True)[len(prompt):])
+# A medium shot of two cats laying on a pile of brown fishing nets. The cat in the foreground is a gray tabby cat with white on its chest and paws. The cat is laying on its side with its head facing the bottom right corner of the image. The cat in the background is laying on its side with its head facing the top left corner of the image. The cat's body is
+```
+
+
+You can also use bitsandbytes to load the models with quantization. The following example uses 4-bit `nf4`:
```
Comment (pcuenca) [inline paligemma2.md:182]:
```suggestion
You can also use the transformers `bitsandbytes` integration to load the models with quantization. The following example uses 4-bit `nf4`:
```

File: paligemma2.md (hunk 8 above)
Comment (pcuenca) [inline paligemma2.md:209]:
```suggestion
If you have previously fine-tuned PaliGemma, the API to fine-tune PaliGemma 2 is the same, you can use your code out of the box. We provide a fine-tuning script and a notebook for you to fine-tune the model, freeze parts of the model, or apply memory efficient fine-tuning techniques like LoRA or QLoRA.
```

File: paligemma2.md [hunk 12]
```diff
@@ -0,0 +1,228 @@
...
+- `8-bit`: 59.78%.
+- `4-bit`, using the configuration from the snippet above: 58.72%.
+
+These are very encouraging figures! Of course, quantization is most interesting for the larger checkpoints, we recommend you always measure results on the domains and tasks you’ll be using.
+
+## Fine-tuning
+
+If you have previously fine-tuned PaliGemma, the API to fine-tune PaliGemma2 is the same, you can use your code out of the box. We provide a fine-tuning script and a notebook for you to fine-tune the model, freeze parts of the model, or apply memory efficient fine-tuning techniques like LoRA or QLoRA.
+
+We have LoRA fine-tuned model on half of the VQAv2 validation split for demonstration purposes. This took half an hour on 3 A100s with 80GB VRAM. The model can be found [here](https://huggingface.co/merve/paligemma2-3b-vqav2) and the demo can be found [here](https://huggingface.co/spaces/merve/paligemma2-vqav2).
```
Comment (pcuenca) [inline paligemma2.md:212]:
```suggestion
We have LoRA-fine-tuned a PaliGemma 2 model on half of the VQAv2 validation split for demonstration purposes. This took half an hour on 3 A100s with 80GB VRAM. The model can be found [here](https://huggingface.co/merve/paligemma2-3b-vqav2), and [this is a Gradio demo that showcases it](https://huggingface.co/spaces/merve/paligemma2-vqav2).
```

File: paligemma2.md (hunk 8 above)
Comment (merveenoyan) [reply paligemma2.md]: adding

File: paligemma2.md [hunk 13]
```diff
@@ -0,0 +1,228 @@
...
+
+
+## Conclusion
+
+The new PaliGemma 2 release is even more exciting than the previous one with various sizes fitting everyone’s needs. We are looking forward to see what the community is going to build off of the even more powerful PaliGemma 2 pre-trained models.
```
Comment (pcuenca) [inline paligemma2.md:217]:
```suggestion
The new PaliGemma 2 release is even more exciting than the previous one, with various sizes fitting everyone’s needs and stronger pre-trained models. We are looking forward to seeing what the community will build!
```

File: paligemma2.md (hunk 7 above)
Comment (pcuenca) [inline paligemma2.md]:
```suggestion
Thanks a lot to the Google team for openly releasing this amazing model family! 🙌 Big thanks to Pablo Montalvo for integrating the model in transformers.
```

File: paligemma2.md (hunk 1 above)
Comment (pcuenca) [reply paligemma2.md:3]: let's use the same here then

File: paligemma2.md [hunk 14]
```diff
@@ -0,0 +1,229 @@
+---
+title: "Welcome PaliGemma 2 – New vision language models by Google"
+thumbnail: /blog/assets/paligemma/paligemma2.png
```
Comment (pcuenca) [inline paligemma2.md:3]:
```suggestion
thumbnail: /blog/assets/paligemma/Paligemma2.png
```

**Review by pcuenca — APPROVED** [review]

**Pull request merged**